nsls2tools/csx1/ophyd/areadetector.py

```diff
@@ -0,0 +1,133 @@
+from ophyd.areadetector.plugins import PluginBase, ProcessPlugin
+from ophyd import Component as Cpt
+from ophyd.device FormattedComponent as FCpt
+from ophyd import AreaDetector
+
+class StandardCam(SingleTrigger, AreaDetector):
+    stats1 = Cpt(StatsPlugin, 'Stats1:')
+    stats2 = Cpt(StatsPlugin, 'Stats2:')
+    stats3 = Cpt(StatsPlugin, 'Stats3:')
+    stats4 = Cpt(StatsPlugin, 'Stats4:')
+    stats5 = Cpt(StatsPlugin, 'Stats5:')
+    roi1 = Cpt(ROIPlugin, 'ROI1:')
+    roi2 = Cpt(ROIPlugin, 'ROI2:')
+    roi3 = Cpt(ROIPlugin, 'ROI3:')
+    roi4 = Cpt(ROIPlugin, 'ROI4:')
+    proc1 = Cpt(ProcessPlugin, 'Proc1:')
+
+
+class NoStatsCam(SingleTrigger, AreaDetector):
+    pass
+
+
+class HDF5PluginSWMR(HDF5Plugin):
+    swmr_active = Cpt(EpicsSignalRO, 'SWMRActive_RBV')
+    swrm_mode = Cpt(EpicsSignalWithRBV, 'SWMRMode')
+    swmr_supported = Cpt(EpicsSignalRO, 'SWMRSupported_RBV')
+    swmr_cb_counter = Cpt(EpicsSignalRO, 'SWMRCbCounter_RBV')
+    _default_configuration_attrs = (HDF5Plugiun._default_configuration_attrs +
+                                    ('swmr_active', 'swrm_mode',
+                                     'swmr_supported'))
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.stage_sigs['swmr_mode'] = 1
+
+
+class HDF5PluginWithFileStore(HDF5PluginSWMR, FileStoreHDF5IterativeWrite):
+    # AD v2.2.0 (at least) does not have this. It is present in v1.9.1.
+    file_number_sync = None
+
+    def get_frames_per_point(self):
+        return self.parent.cam.num_images.get()
+
+class FCCDCam(AreaDetectorCam):
+    sdk_version = Cpt(EpicsSignalRO, 'SDKVersion_RBV')
+    firmware_version = Cpt(EpicsSignalRO, 'FirmwareVersion_RBV')
+    overscan = Cpt(EpicsSignal, 'Overscan')
+    fcric_gain = Cpt(EpicsSignal, 'FCRICGain')
+    fcric_clamp = Cpt(EpicsSignal, 'FCRICClamp')
+    temp = FCpt(EpicsSignal, '{self._temp_pv}')
+
+    def __init__(self, *args, temp_pv=None, **kwargs):
+        self._temp_pv = temp_pv
+        super().__init__(*args, **kwargs)
+
+
+class ProductionCamBase(DetectorBase):
+    # # Trying to add useful info..
+    cam = Cpt(FCCDCam, "cam1:")
+    stats1 = Cpt(StatsPlugin, 'Stats1:')
+    stats2 = Cpt(StatsPlugin, 'Stats2:')
+    stats3 = Cpt(StatsPlugin, 'Stats3:')
+    stats4 = Cpt(StatsPlugin, 'Stats4:')
+    stats5 = Cpt(StatsPlugin, 'Stats5:')
+    roi1 = Cpt(ROIPlugin, 'ROI1:')
+    roi2 = Cpt(ROIPlugin, 'ROI2:')
+    roi3 = Cpt(ROIPlugin, 'ROI3:')
+    roi4 = Cpt(ROIPlugin, 'ROI4:')
+    trans1 = Cpt(TransformPlugin, 'Trans1:')
+    proc1 = Cpt(ProcessPlugin, 'Proc1:')
+
+
+    # This does nothing, but it's the right place to add code to be run
+    # once at instantiation time.
+    def __init__(self, *arg, readout_time=0.04, **kwargs):
+        self.readout_time = readout_time
+        super().__init__(*arg, **kwargs)
+
+    def pause(self):
+        self.cam.acquire.put(0)
+        super().pause()
+
```

Comment:
@ambarb have a look at lines 90 to 100. I am thinking that instead of having these __macros__ which set the FCCD acquisition time, we use this method on the class to do it. Then you would just do:

```python
fccd.set_acquisition(0.1)
```
at the ipython prompt. Curious what @tacaswell and @danielballan think, can this be poked from BS?